.github/workflows/test.yml

```diff
@@ -0,0 +1,47 @@
+name: Test
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```

Comment:
Why is this explicitly limited to `master`?
Won't all (or at least most) PRs be from `master`?
I don't really have experience with non-`master` PRs, but I don't see a reason why to prohibit them. Is there?

Reply:
I think it stems from me trying to limit CI jobs, as for example in private repos you can have only some hundereds each month.
But you are right, we can maybe completely remove the limit here.

Reply:
I removed all limitations and the CI jobs should now be executed for every push / pull request.

Reply:
Now it will run all CI jobs twice for pull requests, one time for the push event, one time for the pull request. This might be not necessary, or would you prefer it this way?

Reply:
If you know a way to avoid the duplication, that would be great. But I think both things individually are very useful.